pulley/src/lib.rs

```diff
@@ -50,13 +46,17 @@ macro_rules! for_each_op {
             /// Branch if unsigned `a <= b`.
             br_if_xulteq32 = BrIfXulteq32 { a: XReg, b: XReg, offset: PcRelOffset };
 
+
+            /// Move between `s` registers.
+            smov = Smov { dst: SReg, src: SReg };
```

Comment:
We shouldn't need this, or any other general instructions for the special registers. They should only be manipulated by instructions that are doing other stuff as well, eg `call` will set `lr`. But these registers aren't allocated by our register allocator and are reserved for their special purpose, so we shouldn't need to have general instructions for them.